odxtools/diaglayer.py

```diff
@@ -636,9 +643,73 @@ def get_comparam(
 
         return cps[0]
 
-    def get_can_receive_id(self, protocol_name: Optional[str] = None) -> Optional[int]:
+    def get_can_mtu(self, protocol: Optional[Union[str, "DiagLayer"]] = None) -> Optional[int]:
+        """Return the maximum size of CAN frames that can be
+        transmitted in bytes.
+
+        For classic CAN busses, this is basically always 8. CAN-FD can
+        send up to 64 bytes per frame.
+        """
+        com_param = self.get_comparam("CP_CANFDTxMaxDataLength", protocol=protocol)
+        if com_param is None:
+            rx_id = self.get_can_receive_id(protocol=protocol)
+            if rx_id is not None:
+                # bus is CAN. We also use 8 bytes for CAN-FD if this
+                # parameter was not specified.
+                return 8
```

Comment:
This is misleading, if the PDX did not explicitly specify an MTU, no MTU should be inferred from other properties

Reply:
`None` always means "Missing" or "Undefined", that's the default fallback when we don't know what the value is.

Reply:
`None` may also know "not applicable", right? if we know it's CAN but not CAN-FD the maximum frame payload size is always 8 bytes. I guess this is what > 90% of people would expect? (I admit that determining whether to use CAN FD is currently quite sketchy, though...)